### DISCUSSION TITLE
Files could not be compiled. Try rebuilding from the source HELP PLEASE!!!

### DISCUSSION THREAD

**Discussion by FlightSimFan94** [JSBSim-Team/jsbsim#1114] [Support requests]
![Image 3-7-2024 at 2 28 PM](https://github.com/JSBSim-Team/jsbsim/assets/174576548/9a0c027a-78be-45c3-8f73-5195427b78f1)
![Image 3-7-2024 at 2 27 PM](https://github.com/JSBSim-Team/jsbsim/assets/174576548/8207350d-a187-4e85-8067-0fca21403582)
Hi, 
I am trying to make my own flight simulator on my macbook air and am very new to unreal engine and all this. Would someone be able to help me fix this problem please?

**Comment by gallonmate** [comment]
You'll need to build the jsbsimlibary for your mac and then compile the unreal plugin for your mac. https://github.com/JSBSim-Team/jsbsim/blob/master/UnrealEngine/README-Unix.md

**Reply by FlightSimFan94** [reply]
<img width="1440" alt="Screenshot 2024-07-04 at 2 25 48 PM" src="https://github.com/JSBSim-Team/jsbsim/assets/174576548/12d35c90-ea41-4d86-b436-cc164def494e">
This is what comes up when i open the JSBSimForUnrealMac.sh
Can you give me a tutorial on how to build the jsbsim library on the mac and compile the plugin. I am using unreal 5.3.2?? Do you have a youtube tutorial or something that can tell me how to do all of it??

**Reply by gallonmate** [reply]
You'll need some software in order to compile the code for mac. But I don't use mac, so I can't help for this part. You should be able to google and find some software that can compile code for you. Perhaps even just finding some guide to setup Unread Engine for C++ development would get you started.
@demonixis or @finger563 may be able to provide some information if you get stuck.

**Reply by demonixis** [reply]
Hey
First you need to install Xcode and Xcode command line tools, then brew.
- In brew install git and git-lfs
- Start XCode, enter credentials and select the default Xcode location (use Xcode.app)
- Then follow the tutorial

If you're new to Unreal and C++, Macos is not the best platform to start with tbh.

@gallonmate is it fine to distribute a compiled version of the plugin on github? Because we can maybe release a precompiled package.

**Reply by FlightSimFan94** [reply]
Hi, 
Thanks for your replies. Sorry it has taken me so long to reply been away from my computer for the past couple weeks.
I have tried to do a bunch of research on softwares to compile the code for my mac but still haven't found anything. 
@demonixis I'll try the xcode tutorial and see if i can get it to work. I would be so appreciative if you could distribute the compiled version of the plugin on github it would greatly help me. 
Many thanks

**Reply by gallonmate** [reply]
@demonixis yup we can distribute it anywhere. If you decide to package the pre-compiled plugin, you could ask @bcoconni if it's possible to add it to the Release section.

**Comment by monurathore** [comment]
@FlightSimFan94 
the master version has some issues so you might able to rebuild it for unreal engine, To build JSB Sim for unreal you have to use the release version. I have tried the 1.2.0 version ((https://github.com/JSBSim-Team/jsbsim/releases/tag/v1.2.0) Try this.
if you face any issues in the rebuild let me know I will share the full process with you.

**Reply by FlightSimFan94** [reply]
Hi @monurathore 
Thanks for your message. I have now downloaded the source code.zip file on the v1.2.0 version. What do I do now?? When I try open the "JSBSimForUnreal.sln" file in Visual Studio 2022, it shows up saying "This project type is not supported by Visual Studio Community for Mac". Is there something I need to do to make it work and build, when i press build in Release x64 in visual studio in the sln file it builds the file straight away but no files are built. Please see attached photo.
![Image 20-7-2024 at 12 11 PM](https://github.com/user-attachments/assets/d292acb6-26e4-4407-9f03-0d0585ad7a91)
Would you be able to share full process with me please??

Many thanks and look forward to hearing from you

**Reply by monurathore** [reply]
Hello,
I have same type of build issues in windows when I was building the sim.
the simplest process is that you have to build the JSB Sim in the release version and then you have to use the plugin in your project, by this you will be able to use the full access off JSB Sim.
Steps for the process: -
Step1. Download the 1.2.0 version of the JSB Sim
Step2. Install the visual studio with all the requirements 
Step3. Locate the file ("JSBSimForUnreal.sln")
Step4. Open the file in visual studio 
Step5. If any requirement changes is required, make the changes.
Step6. Build Solutions
( if the requirement of the file has been full filled then and then only it will be successful, if it is not successful check the requirement then build again.) 
Note:- For all OS there will be different requirements and versions, so you have to find the suitable requirements  by testing it with different versions) ( It took me a while to find the prefect version)
Step7. if the build is successful then make a project file with source 
Step8. Copy the plugin into your project file and generate Visual Studio Project File 
Step9. Open the visual studio project file and Build the project file 
(If build is successful you can run your file)
in the complete process the time taking is step 6 and 9 
this two process will take a lot of time and not easily successful.
But I have used the same process and it is working.
best of luck, If you need any help let me know.
Thank You
